docs/spark/how-to-guides/windows-instructions.md

```diff
@@ -0,0 +1,253 @@
+     - Verify you are able to run `spark-shell` from your command-line.        
+        Sample console output:
+
+        ```
+        Welcome to
+              ____              __
+             / __/__  ___ _____/ /__
+            _\ \/ _ \/ _ `/ __/  '_/
+           /___/ .__/\_,_/_/ /_/\_\   version 2.3.2
+              /_/
+
+        Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201)
+        Type in expressions to have them evaluated.
+        Type :help for more information.
+
+        scala> sc
+        res0: org.apache.spark.SparkContext = org.apache.spark.SparkContext@6eaa6b0c
+        ```
+
+        </details>
+
+  6. Install **[WinUtils](https://github.com/steveloughran/winutils)**.
+     - Download `winutils.exe` binary from [WinUtils repository](https://github.com/steveloughran/winutils). You should select the version of Hadoop the Spark distribution was compiled with, e.g. use hadoop-2.7.1 for Spark 2.3.2.
+     - Save `winutils.exe` binary to a directory of your choice e.g., `c:\hadoop\bin`.
+     - Set `HADOOP_HOME` to reflect the directory with winutils.exe (without bin). For instance, using command-line:
+
+       ```powershell
+       set HADOOP_HOME=c:\hadoop
+       ```
+
+     - Set PATH environment variable to include `%HADOOP_HOME%\bin`. For instance, using command-line:
+
+       ```powershell
+       set PATH=%HADOOP_HOME%\bin;%PATH%
+       ```
+
+Make sure you are able to run `dotnet`, `java`, `mvn`, `spark-shell` from your command-line before you move to the next section. Feel there is a better way? Please [open an issue](https://github.com/dotnet/spark/issues) and feel free to contribute.
+
+> [!NOTE]
+> A new instance of the command-line may be required if any environment variables were updated.
+
+## Build
+
+For the rest of the section, it is assumed that you have cloned the .NET for Apache Spark repository into your machine e.g., `c:\github\dotnet-spark\`.
+
+```bash
+git clone https://github.com/dotnet/spark.git c:\github\dotnet-spark
+```
+
+### Build Spark .NET Scala extensions layer
+
+When you submit a .NET application, Spark .NET has the necessary logic written in Scala that inform Apache Spark how to handle your requests (e.g., request to create a new Spark Session, request to transfer data from .NET side to JVM side etc.). This logic can be found in the [Spark .NET Scala Source Code](https://github.com/dotnet/spark/tree/master/src/scala).
+
+Regardless of whether you are using .NET Framework or .NET Core, you will need to build the Spark .NET Scala extension layer. This is easy to do:
+
+```powershell
+cd src\scala
+mvn clean package 
+```
+
+You should see JARs created for the supported Spark versions:
+
+* `microsoft-spark-2.3.x\target\microsoft-spark-2.3.x-<version>.jar`
+* `microsoft-spark-2.4.x\target\microsoft-spark-2.4.x-<version>.jar`
+
+### Build a .NET samples application
```

Comment:
Could we provide a sentence/short paragraph about what this means? i.e. Introduce and link to the samples.

Reply:
Also perhaps reword this title? Something like "Build a sample application" or "Build the .NET for Spark sample apps"

Reply:
How about ->
This section explains how to build the [sample applications](*link to apps in Spark .NET github repo*) for .NET for Apache Spark. These steps will help in understanding the overall building process for any .NET for Spark application.